lib/auth_page.dart

```diff
@@ -66,6 +83,56 @@ Stream<int> _signin(BuildContext context) async* {
   }
 }
 
+void _facebookSignout() {
+  try {
+    final _facebookLogin = FacebookLogin();
+    _facebookLogin.logOut();
+  } catch (error, trace) {
+    // reset the UI and display an alert
+  }
+}
+
+Stream<int> _facebookSignin(BuildContext context) async* {
+  try {
+    final _fireAuth = FirebaseAuth.instance;
+    final facebookLogin = FacebookLogin();
+    final result = await facebookLogin.logIn(['email']);
+
+    switch (result.status) {
+      case FacebookLoginStatus.loggedIn:
+        /// the auth info will be picked up by the listener on [onAuthStateChanged]
+        /// and emitted by [streamOfStateChanges]
+        
+        // signal to change UI
+        yield 2;
+
+        final credential = FacebookAuthProvider.getCredential(accessToken: result.accessToken.token);
+        await _fireAuth.signInWithCredential(credential);
+
+        // we are signed in so reset the UI
+        yield 0;
+        break;
+      case FacebookLoginStatus.cancelledByUser:
```

Comment:
In the case where the user cancels, just resetting the UI seems appropriate so my opinion is let's get rid of the `// _showCancelledMessage();` comment

Reply:
I'll be honest I didn't really understand what the Google was doing with yield 0, and yield 2 etc. I just followed the pattern google had set.

Does dart have enums?

Reply:
I think the stream of ints was from my code maybe?  That was my terrible way of doing it.  Absolutely agree we should use enums (and yep dart has enums, they're a bit limited but they're there - there are a bunch of package solutions too, eg. built_value has class enums).  I think in my case I had the UI reset whenever the stream emitted a 0 (it was like a reset I think).  

Anyway the only point I was trying to make was let's get rid of the comment `// _showCancelledMessage();` as I don't think there's going to be any reason to display a message when the user cancels.  Happy to be disagreed with :-)